src/Util/App.cc

```diff
@@ -33,3 +42,40 @@ bool FindExecutablePath(std::string& path) {
     path = path_buffer;
     return true;
 }
+
+std::string GetReleaseInformation() {
+#ifdef __APPLE__
+    // MacOS solution adapted from https://stackoverflow.com/a/44684199/1727322
+    char info_buffer[MAX_PLATFORM_INFO_LENGTH];
+    unsigned buffer_length = 0;
+    char line[MAX_PLATFORM_LINE_LENGTH];
+    auto file_handle = popen("sw_vers", "r");
+    while (fgets(line, sizeof(line), file_handle) != nullptr) {
+        int l = snprintf(info_buffer + buffer_length, sizeof(info_buffer) - buffer_length, "%s", line);
+        buffer_length += l;
+        if (buffer_length > MAX_PLATFORM_INFO_LENGTH) {
+            spdlog::warn("Problem reading platform information");
+            return std::string("Platform information not available");
+        }
+    }
+    pclose(file_handle);
+    return info_buffer;
```

Comment:
I considered doing this, but I'd rather handle the platform interpretation on the telemetry server side of things. That way, if Apple changes the output of `sw_vers` in some way, we can adapt our server-side code, instead of needing to push a fix to every CARTA instance in the world.

The approach I've taken for MacOS is somewhat consistent with the Linux approach. `sw_vers` output looks like

```
ProductName:    Mac OS X
ProductVersion: 10.12.5
BuildVersion:   16F73
```

and `/etc/os-release` looks like

```
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

both will require some regex work on the server-side, but that's much easier to iterate on than altering CARTA itself.